File: WorkSpace/Spark-Projects/src/main/java/spark/sql/loadSave/ManuallySpecifyOptions.java
package spark.sql.loadSave;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

/**
 * 手动指定数据源类型
 * @author Administrator
 *
 */
public class ManuallySpecifyOptions {

	public static void main(String[] args) {
		SparkConf conf = new SparkConf()   
				.setAppName("ManuallySpecifyOptions");
		JavaSparkContext sc = new JavaSparkContext(conf);
		SQLContext sqlContext = new SQLContext(sc);
		
		DataFrame peopleDF = sqlContext.read().format("json")
				.load("hdfs://spark1:9000/people.json");
		peopleDF.select("name").write().format("parquet")  
				.save("hdfs://spark1:9000/peopleName_java");     
	}
	
}

File: WorkSpace/Spark-Projects/src/main/scala/spark1/sql/loadSave/ManuallySpecifyOptions.scala
package spark1.sql.loadSave

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.sql.SQLContext

/**
 * @author Administrator
 */
object ManuallySpecifyOptions {
  
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
        .setAppName("ManuallySpecifyOptions")  
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)
  
    val peopleDF = sqlContext.read.format("json").load("hdfs://spark1:9000/people.json")
    peopleDF.select("name").write.format("parquet").save("hdfs://spark1:9000/peopleName_scala")   
  }
  
}